src/main/java/uk/gov/companieshouse/itemhandler/service/EmailService.java

```diff
@@ -103,56 +51,34 @@ public EmailService(
+            emailSend.setMessageId(UUID.randomUUID().toString());
+            emailSend.setCreatedAt(LocalDateTime.now().toString());
 
-                String orderReference = itemGroup.getOrder().getReference();
-                LoggingUtils.logWithOrderReference("Sending confirmation email for order", orderReference);
-                emailSendProducer.sendMessage(emailSend, orderReference);
-            }
+            String orderReference = itemGroup.getOrder().getReference();
+            LoggingUtils.logWithOrderReference("Sending confirmation email for order", orderReference);
+            emailSendProducer.sendMessage(emailSend, orderReference);
         } catch (JsonProcessingException exception) {
             String msg = String.format("Error converting order (%s) confirmation to JSON", itemGroup.getOrder().getReference());
             LOGGER.error(msg, exception);
             throw new NonRetryableException(msg);
         }
     }
 
-    /**
-     * Builds the order confirmation and email based on the order provided.
-     * @param order the order for which an email confirmation is to be sent
-     * @return a {@link OrderConfirmationAndEmail} holding both the confirmation and its email envelope
-     */
-    private OrderConfirmationAndEmail buildOrderConfirmationAndEmail(final OrderData order) {
-        final String deliveryTimescale = ((DeliveryItemOptions) order.getItems().get(0).getItemOptions()).getDeliveryTimescale().getJsonName();
-        final EmailSend email = new EmailSend();
-        final OrderConfirmation confirmation;
-        confirmation = orderToItemOrderConfirmationMapper.orderToConfirmation(order);
-        confirmation.setTo(certifiedCopyOrderRecipient);
-        email.setAppId(deliveryTimescale.equals(STANDARD_DELIVERY) ?
-                CERTIFIED_COPY_ORDER_NOTIFICATION_API_APP_ID :
-                SAME_DAY_CERTIFIED_COPY_ORDER_NOTIFICATION_API_APP_ID);
-        email.setMessageType(deliveryTimescale.equals(STANDARD_DELIVERY) ?
-                CERTIFIED_COPY_ORDER_NOTIFICATION_API_MESSAGE_TYPE :
-                SAME_DAY_CERTIFIED_COPY_ORDER_NOTIFICATION_API_MESSAGE_TYPE);
-        return new OrderConfirmationAndEmail(confirmation, email);
+    private EmailSend createEmailSendBasedOnKind(DeliverableItemGroup itemGroup) throws JsonProcessingException {
+        EmailSend emailSend = new EmailSend();
+        if (ITEM_KIND_CERTIFIED_COPY.equals(itemGroup.getKind())) {
+            EmailMetadata<CertifiedCopyEmailData> emailMetadata = confirmationMapperFactory.getCertifiedCopyMapper().map(itemGroup);
+            emailSend.setAppId(emailMetadata.getAppId());
+            emailSend.setMessageType(emailMetadata.getMessageType());
+            emailSend.setData(objectMapper.writeValueAsString(emailMetadata.getEmailData()));
+        } else {
+            EmailMetadata<CertificateEmailData> emailMetadata = confirmationMapperFactory.getCertificateMapper().map(itemGroup);
```

Comment:
Duplicate code; you'd be better off moving logic back into `sendOrderConfirmation` and passing in the mapper as an argument:

```java
EmailSend emailModel;
if (ITEM_KIND_CERTIFIED_COPY.equals(itemGroup.getKind())) {
    emailModel = mapEmailSend(itemGroup, confirmationMapperFactory.getCertifiedCopyMapper());
else if (ITEM_KIND_CERTIFICATE.equals(itemGroup.getKind())) {
    emailModel = mapEmailSend(itemGroup, confirmationMapperFactory.getCertificateMapper());
} else {
    throw new RuntimeException(String.format("Unknown item kind: [%s]", itemGroup.getKind()));
}
...
```

```java
private EmailSend createEmailSendBasedOnKind(DeliverableItemGroup itemGroup) throws JsonProcessingException {
    EmailSend emailSend = new EmailSend();
    EmailMetadata<CertifiedCopyEmailData> emailMetadata = confirmationMapperFactory.getCertifiedCopyMapper().map(itemGroup);
    emailSend.setAppId(emailMetadata.getAppId());
    emailSend.setMessageType(emailMetadata.getMessageType());
    emailSend.setData(objectMapper.writeValueAsString(emailMetadata.getEmailData()));
    emailSend.setEmailAddress(TOKEN_EMAIL_ADDRESS);
    emailSend.setMessageId(UUID.randomUUID().toString());
    emailSend.setCreatedAt(LocalDateTime.now().toString());
    return emailSend;
}
```

Reply:
Resolved in latest commit.